applauncher/004-three-by-three.qml

```diff
@@ -204,9 +145,60 @@ Item {
                 }
             }
         }
+    }
+
+    Component {
+        id: spacer
+        Item {
+            height: root.height / numColumns / 2
+            width: height
+        }
+    }
+
+    GridView {
+        id: appsView
+
+        flow: GridView.FlowLeftToRight
+        snapMode: GridView.SnapToRow
+        anchors {
+            fill: parent
+            leftMargin: parent.width * .026
+            rightMargin: parent.width * .026
+        }
+        clip: true
+        cellHeight: height / numColumns
+        cellWidth: width / numColumns
+        contentY: -height / (2*numColumns)
+
+        header: spacer
+        footer: spacer
+
+        property int currentPos: 0
+
+        onCurrentPosChanged: {
+            rightIndicator.animate()
+            leftIndicator.animate()
+            topIndicator.animate()
+            bottomIndicator.animate()
+        }
+
+        onDragStarted: {
+            dragStop = true
+        }
+
+        onAtYBeginningChanged: {
+            // Make sure that the grid doesn't move when the app view is visible.
+            if ((grid.currentHorizontalPos === 0) && (grid.currentVerticalPos === 1)) {
+                forbidTop = !atYBeginning
+                grid.changeAllowedDirections()
+            }
+        }
+
+        model: launcherModel
+
+        delegate: gridDelegate
 
         Component.onCompleted: {
-            launcherColorOverride = true
```

Comment:
This is not dead code, this is affecting the behavior of https://github.com/AsteroidOS/asteroid-launcher/blob/d5f4f7361b9896fedbc03e575c2576b6fac74bab/qml/MainScreen.qml#L364

I know... This is a pretty terrible pattern to have these implicit globally defined variables :( Suggestions welcome

Reply:
I'll add a comment to note variables which are from the parent context.  There's probably a better way to do it, but I can't think of it at the moment.

Reply:
Also, when selecting a launcher, this variable is not available from within `asteroid-settings` so the result is an error message (for each launcher):   
```
Mar 28 10:06:55 catfish asteroid-launcher[2170]: file:///usr/share/asteroid-launcher/applauncher/004-three-by-three.qml:216: Error: Invalid write to global property "launcherColorOverride"
```
But that's an issue for `asteroid-settings` so I will add another PR there.